lakey_service/downloader/executors/athena.py

```diff
@@ -1,4 +1,242 @@
 
+from io import StringIO
+from time import sleep
+
+from django.conf import settings
+import boto3
+import requests
+import pandas as pd
+
+
+athena = boto3.client(
+    'athena',
+    region_name=settings.AWS_LAKEY_REGION,
+    aws_access_key_id=settings.AWS_LAKEY_KEY_ID,
+    aws_secret_access_key=settings.AWS_LAKEY_KEY_SECRET)
+
+
+s3 = boto3.client(
+    's3',
+    region_name=settings.AWS_LAKEY_REGION,
+    aws_access_key_id=settings.AWS_LAKEY_KEY_ID,
+    aws_secret_access_key=settings.AWS_LAKEY_KEY_SECRET)
+
 
 class AthenaExecutor:
-    pass
+
+    def execute(self, download_request):
+
+        query = self.compile_to_query(download_request)
+
+        database = download_request.catalogue_item.name.split('.')[0]
+
+        return self.execute_query(database, query)
+
+    def compile_to_query(self, download_request):
+
+        table = download_request.catalogue_item.name
+        spec = download_request.spec
+
+        # -- COLUMNS
+        columns = ', '.join(spec['columns'])
+
+        # -- FILTERS
+        filters_entries = []
+        for f in spec['filters']:
```

Comment:
i would use different variable named becouse few lines below it is mixed with fstring and it can be confusing

Reply:
of course I will make that change